packages/cli/src/ui/components/Composer.tsx

```diff
@@ -0,0 +1,154 @@
+/**
+ * @license
+ * Copyright 2025 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React from 'react';
+import { Box, Text } from 'ink';
+import { LoadingIndicator } from './LoadingIndicator.js';
+import { ContextSummaryDisplay } from './ContextSummaryDisplay.js';
+import { AutoAcceptIndicator } from './AutoAcceptIndicator.js';
+import { ShellModeIndicator } from './ShellModeIndicator.js';
+import { DetailedMessagesDisplay } from './DetailedMessagesDisplay.js';
+import { InputPrompt } from './InputPrompt.js';
+import { Footer, FooterProps } from './Footer.js';
+import { ShowMoreLines } from './ShowMoreLines.js';
+import { OverflowProvider } from '../contexts/OverflowContext.js';
+import { Colors } from '../colors.js';
+import { isNarrowWidth } from '../utils/isNarrowWidth.js';
+import { useUIState } from '../contexts/UIStateContext.js';
+import { useUIActions } from '../contexts/UIActionsContext.js';
+import { useVimMode } from '../contexts/VimModeContext.js';
+import { useConfig } from '../contexts/ConfigContext.js';
+import { ApprovalMode } from '@google/gemini-cli-core';
+import { StreamingState } from '../types.js';
+
+interface ComposerProps {
+  contextFileNames: string[];
+  showAutoAcceptIndicator: ApprovalMode;
+  footerProps: Omit<FooterProps, 'vimMode'>;
+}
+
+export const Composer = (props: ComposerProps) => {
+  const { contextFileNames, showAutoAcceptIndicator, footerProps } = props;
+
+  const config = useConfig();
+
+  const uiState = useUIState();
```

Comment:
this is a case where the uiState generally makes sense. I would suggest we pass it in as a property instead of using Context.